rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositorySpec.scala

```diff
@@ -88,6 +88,24 @@ class HistoryRepositorySpec
     } yield ()
   }
 
+  it should "not allow switching to a not existing root" in withEmptyRepository { repo =>
+    Task.delay {
+      the[RuntimeException] thrownBy {
+        repo.reset(zerosBlake).runSyncUnsafe(20.seconds)
```

Comment:
Something similar to this should work:
```scala
repo.reset(zerosBlake).attempt.map {
  case Left(RuntimeException("unknown root")) => succeed()
  case _ => fail("Expected a failure")
}
```

Reply:
`RuntimeException` does not have unapply :(